var/spack/repos/builtin/packages/curl/package.py

```diff
@@ -48,15 +48,18 @@ class Curl(AutotoolsPackage):
     version('7.43.0', '11bddbb452a8b766b932f859aaeeed39')
     version('7.42.1', '296945012ce647b94083ed427c1877a8')
 
+    variant('nghttp2', default=False, description='build nghttp2 library (requires C++11)')
```

Comment:
Would this variant be better named "http2" and desc='Enable http2 support (via nghttp2 lib, requires C++11)' since the use of nghttp2 is an implementation detail and the added capability is the ability to do HTTP/2?

Reply:
I think the variant name must match the flag name in order to use `with_or_without`.